Content.Server/Objectives/Conditions/HijackShuttleCondition.cs

```diff
@@ -0,0 +1,122 @@
+﻿using Content.Server.Mind;
+using Content.Server.Objectives.Interfaces;
+using Content.Server.Roles;
+using Content.Server.Shuttles.Components;
+using Content.Shared.Cuffs.Components;
+using Robust.Shared.Player;
+using Robust.Shared.Utility;
+
+namespace Content.Server.Objectives.Conditions
+{
+    [DataDefinition]
+    public sealed partial class HijackShuttleCondition : IObjectiveCondition
+    {
+        private MindComponent _mind;
+        private EntityUid _mindId;
+
+        public IObjectiveCondition GetAssigned(EntityUid mindId, MindComponent mind)
+        {
+            return new HijackShuttleCondition
+            {
+                _mind = mind,
+                _mindId = mindId,
+            };
+        }
+
+        public string Title => Loc.GetString("objective-condition-hijack-shuttle-title");
+
+        public string Description => Loc.GetString("objective-condition-hijack-shuttle-description");
+
+        public SpriteSpecifier Icon => new SpriteSpecifier.Rsi(new ResPath("Objects/Tools/emag.rsi"), "icon");
+
+        private bool IsShuttleHijacked(EntityUid shuttleGridId)
+        {
+            var entMan = IoCManager.Resolve<IEntityManager>();
+            var mindSystem = entMan.System<MindSystem>();
+            var roleSystem = entMan.System<RoleSystem>();
+
+            var agentOnShuttle = false;
+            var gridPlayers = Filter.BroadcastGrid(shuttleGridId).Recipients;
+            foreach (var player in gridPlayers)
+            {
+                if (!player.AttachedEntity.HasValue ||
+                    !mindSystem.TryGetMind(player.AttachedEntity.Value, out var mindId, out var mind))
+                    continue;
+
+                if (mindId == _mindId)
+                {
+                    agentOnShuttle = true;
+                    continue;
+                }
+
+                var isPersonTraitor = roleSystem.MindHasRole<TraitorRoleComponent>(mindId);
+                if (isPersonTraitor) // Allow traitors
+                    continue;
+
+                var isPersonDead = mindSystem.IsCharacterDeadIc(mind);
```

Comment:
I think this should also count people in crit. Otherwise you would need to ensure that every non-traitor role on shuttle is dead and finish them off if they not. That will consume a lot of time that you mainly need to deal with everyone who will try to stop you.

Also it looks like code doesn't exclude non-humanoid players, so any mouse, hampter, pAI or even revenant under player control could ruin this objective